core/model/src/main/java/org/eclipse/rdf4j/model/impl/DynamicModel.java

```diff
@@ -141,6 +142,35 @@ public boolean remove(Resource subj, IRI pred, Value obj, Resource... contexts)
 
 	@Override
 	public Model filter(Resource subj, IRI pred, Value obj, Resource... contexts) {
+		if (model == null && statements.size() == 0) {
+			return new EmptyModel(this);
```

Comment:
This isn't valid - you can't return a new Model object as the result of the filter. If someone were to add something to this empty model, that should be reflected in the original.

Reply:
I'll push a couple of tests for this to this PR to make it easier.